cumulusci/tasks/metadeploy.py

```diff
@@ -0,0 +1,123 @@
+import json
+import requests
+
+from cumulusci.core.config import BaseConfig
+from cumulusci.core.exceptions import PlanNotFound
+from cumulusci.core.tasks import BaseTask
+from cumulusci.core.flowrunner import FlowCoordinator
+
+
+class Publish(BaseTask):
+    """Publishes an installation plan to MetaDeploy.
+    """
+
+    task_options = {
+        "plan": {
+            "description": "Name of the installation plan to publish",
+            "required": True,
+        },
+        "tag": {"description": "Name of git tag to publish"},
+    }
+
+    def _init_task(self):
+        metadeploy_service = self.project_config.keychain.get_service("metadeploy")
+        self.base_url = metadeploy_service.url
+        self.api = requests.Session()
+        self.api.headers["Authorization"] = "token {}".format(metadeploy_service.token)
+        self.metadeploy_config = self.project_config.metadeploy
+        if self.metadeploy_config is None:
+            raise PlanNotFound(
+                "No metadeploy configuration found. "
+                "Did you forget to configure it in cumulusci.yml?"
+            )
+        plan_config = getattr(
```

Comment:
i've tried to encourage us to get away from this habit of reaching deep into project_config in task classes. instead, you can make each thing from the metadeploy_config that you actually need as a task_option, and then give them an option default using the $project_config option syntax.

Reply:
At first glance that sounds quite reasonable, I'll see how it plays out.